### ISSUE TITLE
[L4D2 - "Bug"*] Downloading Many Add-ons Floods the Home Folder

### ISSUE THREAD

**Issue by ThisNekoGuy** [ValveSoftware/steam-for-linux#8537]
### **Software and Hardware:**
OS / Distro: Arch Linux
Kernel: [Linux 5.17.2-256-tkg-pds-llvm](https://github.com/frogging-family/linux-tkg)
Desktop Environment: KDE Plasma 5.24.4
Window Manager: Kwin
GPU: Nvidia RTX 2080 Ti
GPU Driver: 510.60.02
FIlesystem in use: BTRFS

### **Description:**
When downloading add-ons for L4D2, it downloads them to my OS drive **_instead of the storage drive I have L4D2 installed on_**, causing me to run out of storage where I desperately need it.
Feel free to ask any questions that would assist in understanding if that isn't clear, but I don't think that's how downloading add-ons for the game should work. :|

While not strictly a bug, it can be quite a severe issue

[System Info](https://gist.github.com/ThisNekoGuy/e03e32d6a29a46ae86ee49e4193331f2)

**Comment by kisak-valve** [comment]
Hello @ThisNekoGuy, this sounds like you're referring to workshop content for Left 4 Dead 2, which is then referred to as add-ons in-game. The download and management of workshop content in general is handled by the Steam client, not the game, so I've transferred this issue report. It also sounds like the Steam client is choosing to use `~/.local/share/Steam/steamapps/workshop/content/550/` instead of the Steam library you expected.

**Comment by ThisNekoGuy** [comment]
That is correct; that makes sense, since they can be subscribed to via the client, but they should honestly be allocated more appropriately, namely to the storage drive the game is installed on...

**Comment by ThisNekoGuy** [comment]
Not sure if this is directly related or not but I had Sniper Elite 5 installed and, I tried to update the game, but it decided that downloading all of the updated data to my OS drive was appropriate when it wasn't: I just now ended up running out of storage space and had no idea why until I examined my Home folder with [filelight](https://apps.kde.org/filelight/)
